src/cmd/device-protection.js

```diff
@@ -325,25 +303,26 @@ module.exports = class DeviceProtectionCommands extends CLICommandBase {
 	 * @param {Function} fn - The function to execute with the device.
 	 * @returns {Promise<*>} The result of the function execution.
 	 */
-	async _withDevice({ putDeviceBackInDfuMode, spinner }, fn) {
-		try {
-			await this._getUsbDevice(this.device);
-			return await this.ui.showBusySpinnerUntilResolved(spinner, (async () => {
-				const deviceWasInDfuMode = this.device.isInDfuMode;
-				if (deviceWasInDfuMode) {
+	async _withDevice({ putDeviceBackInDfuMode, spinner, supportSafeMode }, fn) {
+		await this._getUsbDevice(this.device);
+		await this.ui.showBusySpinnerUntilResolved(spinner, (async () => {
+			this.status = await DeviceProtectionHelper.getProtectionStatus(this.device);
+			const deviceWasInDfuMode = this.device.isInDfuMode;
+			if (deviceWasInDfuMode) {
+				if (!this.status.protected || supportSafeMode) {
 					await this._putDeviceInSafeMode();
+					this.status = await DeviceProtectionHelper.getProtectionStatus(this.device);
 				}
```

Comment:
I'd very much like to avoid this behavior because it makes any command in DFU reset the device a couple of times.